zrml/swaps/src/lib.rs

```diff
@@ -1198,6 +1198,70 @@ mod pallet {
                 .ok_or(Error::<T>::AssetNotBound)
         }
 
+        fn set_pool_as_stale_common(pool_id: PoolId) -> Result<Weight, DispatchError> {
+            Self::mutate_pool(pool_id, |pool| {
+                if pool.pool_status == PoolStatus::Stale {
+                    return Ok(());
+                }
+                ensure!(pool.pool_status == PoolStatus::Active, Error::<T>::InvalidStateTransition);
+                pool.pool_status = PoolStatus::Stale;
+                Ok(())
+            })?;
+            Ok(0) // TODO Needs benchmark!
+        }
```

Comment:
We can get a quite precise estimation by relying on the biggest chunk of weight in this function: DB reads and writes.

```suggestion
        fn set_pool_as_stale_common(pool_id: PoolId) -> Result<Weight, DispatchError> {
            let mut total_weight = T::DbWeight::get().reads(1);

            Self::mutate_pool(pool_id, |pool| {
                if pool.pool_status == PoolStatus::Stale {
                    return Ok(());
                }
                ensure!(pool.pool_status == PoolStatus::Active, Error::<T>::InvalidStateTransition);
                pool.pool_status = PoolStatus::Stale;
                total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
                Ok(())
            })?;

            Ok(total_weight)
        }
```

I am also thinking about allowing the function only to be called on `Active` state to enforce only valid state transitions. This should uncover errors - the market should not try to set the same pool to stale twice, if this happens something unplanned happened

```suggestion
        fn set_pool_as_stale_common(pool_id: PoolId) -> Result<Weight, DispatchError> {
            Self::mutate_pool(pool_id, |pool| {
                ensure!(pool.pool_status == PoolStatus::Active, Error::<T>::InvalidStateTransition);
                pool.pool_status = PoolStatus::Stale;
                Ok(())
            })?;

            Ok(T::DbWeight::get().reads_writes(1, 1))
        }
```